integration/command.go

```diff
@@ -175,23 +183,26 @@ func (c *command) start(t *testing.T) {
 // Cmd output is then checked with regard to expectedString and expectedRegexp
 func (c *command) stop(t *testing.T) {
 	t.Logf("Stop command: %s\n", c.cmd)
-	err := c.command.Process.Kill()
+
+	// Here, we just need to send the PID of /bin/sh (which is the same PGID) as
+	// a negative number to syscall.Kill(). As a result, the signal will be
+	// received by all the processes with such PGID, in our case, the process of
+	// /bin/sh and c.cmd.
+	err := syscall.Kill(-c.command.Process.Pid, syscall.SIGTERM)
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	stdout := c.stdout.String()
 	stderr := c.stderr.String()
+	t.Logf("command returned:\n%s\n%s\n", stderr, stdout)
 
 	if c.expectedRegexp != "" {
 		r := regexp.MustCompile(c.expectedRegexp)
 		if !r.MatchString(stdout) {
-			fmt.Fprintf(os.Stderr, "%s\n", stderr)
 			t.Fatalf("regexp didn't match: %s\n%s\n", c.expectedRegexp, stdout)
 		}
-	}
-	if c.expectedString != "" && stdout != c.expectedString {
-		fmt.Fprintf(os.Stderr, "%s\n", stderr)
+	} else if c.expectedString != "" && stdout != c.expectedString {
```

Comment:
I'm not sure if we really want to change this behavior. I think it's better to have two independent conditionals in the (extreme) case somebody sets both, `expectedRegexp` and `expectedString`. Otherwise one of them will be ignored silently.

Reply:
Yeah, for me it is also "extreme", I do not expect them to be used together but it's ok. I will keep them independent.